actors/sunbird-lms-mw/actors/common/src/main/java/org/sunbird/learner/actors/qrcodedownload/QRCodeDownloadManagementActor.java

```diff
@@ -0,0 +1,223 @@
+        ExecutionContext.setRequestId(request.getRequestId());
+        String requestedOperation = request.getOperation();
+        switch (requestedOperation) {
+            case "downloadQRCodes":
+                downloadQRCodes(request);
+                break;
+
+            default:
+                onReceiveUnsupportedOperation(requestedOperation);
+                break;
+        }
+    }
+
+    /**
+     * The request must contain list of userIds (Users Ids of people who have created courses)
+     * @param request
+     */
+    private void downloadQRCodes(Request request) {
+        Map<String, String> headers = (Map<String, String>) request.getRequest().get(JsonKey.HEADER);
+        Map<String, Object> requestMap = (Map<String, Object>) request.getRequest().get(JsonKey.FILTER);
+        requestMap.put(JsonKey.CONTENT_TYPE, "course");
+        Map<String, Object> searchResponse = searchCourses(requestMap, headers);
+        List<Map<String, Object>> contents = (List<Map<String, Object>>) searchResponse.get("contents");
+        if (CollectionUtils.isEmpty(contents))
+            throw new ProjectCommonException(
+                    ResponseCode.errorUserHasNotCreatedAnyCourse.getErrorCode(),
+                    ResponseCode.errorUserHasNotCreatedAnyCourse.getErrorMessage(),
+                    ResponseCode.CLIENT_ERROR.getResponseCode());
+        Map<String, List<String>> dialCodesMap = contents.stream()
+                .filter(content -> content.get("dialcodes") != null)
+                .filter(content -> content.get("name") != null)
+                .collect(Collectors.toMap(content -> ((String) content.get("identifier")) + "<<<" + (String) content.get("name"), content -> (List) content.get("dialcodes")));
+        File file = generateCSVFile(dialCodesMap);
+        Response response = new Response();
+        if (null == file)
+            throw new ProjectCommonException(
+                    ResponseCode.errorProcessingFile.getErrorCode(),
+                    ResponseCode.errorProcessingFile.getErrorMessage(),
+                    ResponseCode.SERVER_ERROR.getResponseCode());
+
+        response = uploadToAws(file);
+        response.put(JsonKey.USER_IDs, requestMap.get("userIds"));
+        sender().tell(response, self());
+    }
+
+    /**
+     * Search call to LP composite search engine
+     * @param requestMap
+     * @param headers
+     * @return
+     */
+    private Map<String, Object> searchCourses(Map<String, Object> requestMap, Map<String, String> headers) {
+        String request = prepareSearchRequest(requestMap);
+        Map<String, Object> searchResponse = ContentSearchUtil.searchContentSync(null, request, headers);
+        return searchResponse;
+    }
+
+    /**
+     * Request Preparation for search Request
+     * @param requestMap
+     * @return
+     */
+    private String prepareSearchRequest(Map<String, Object> requestMap) {
```

Comment:
Since this is a private method and is used within this class, we can specifically prepare the search request for course search with dialcodes

Reply:
This is required for future usage.